pkg/kv/kvclient/kvcoord/txn_coord_sender.go

```diff
@@ -698,8 +698,12 @@ func (tc *TxnCoordSender) maybeRejectClientLocked(
 	case txnError:
 		return tc.mu.storedErr
 	case txnFinalized:
+		summary := "nil batch"
+		if ba != nil {
+			summary = ba.Summary()
+		}
```

Comment:
Oops, thanks! Maybe it'd be better to just make `Summary()` safe to use with a `nil` receiver? No strong opinion either way, this works too.

Reply:
yeah i will make it handle a nil receiver. i wasn't sure which KV would find more conventional